snorkel/analysis/metrics.py

```diff
@@ -91,14 +91,21 @@ def _roc_auc_score(golds: np.ndarray, probs: np.ndarray) -> float:
     return skmetrics.roc_auc_score(golds, probs[:, 0])
 
 
+def _f1_score(golds: np.ndarray, preds: np.ndarray) -> float:
+    if golds.max() <= 1:
+        return skmetrics.f1_score(golds, preds)
+    else:
+        return skmetrics.f1_score(golds, preds, average="micro")
+
+
 # See https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics
 # for details on the definitions and available kwargs for all metrics from scikit-learn
 METRICS = {
     "accuracy": Metric(skmetrics.accuracy_score),
     "coverage": Metric(_coverage_score, ["preds"]),
     "precision": Metric(skmetrics.precision_score),
     "recall": Metric(skmetrics.recall_score),
-    "f1": Metric(skmetrics.f1_score),
+    "f1": Metric(_f1_score, ["golds", "preds"]),
```

Comment:
if we're hardcoding micro, can we have 3 entries: "f1" that only does skmetrics.f1_score, "f1_micro" that does skmetrics.f1_score(golds, preds, average="micro"), and "f1_macro" that does skmetrics.f1_score(golds, preds, average="macro")?

Reply:
done